universe/interface.go

```diff
@@ -84,6 +84,66 @@ func (i *Identifier) StringForLog() string {
 		i.String(), i.AssetID[:], groupKey, i.ProofType)
 }
 
+// NewUniIDFromRawArgs creates a new universe ID from the raw arguments. The
+// asset ID bytes and group key bytes are mutually exclusive. If the group key
+// bytes are set, then the asset ID bytes will be ignored.
+// This function is useful in deriving a universe ID from the data stored in the
+// database.
+func NewUniIDFromRawArgs(assetIDBytes []byte, groupKeyBytes []byte,
+	proofTypeStr string) (Identifier, error) {
+
+	proofType, err := ParseStrProofType(proofTypeStr)
+	if err != nil {
+		return Identifier{}, err
+	}
+
+	// If the group key bytes are set, then we'll preferentially populate
+	// the universe ID with that and not the asset ID.
+	if groupKeyBytes != nil {
```

Comment:
Should always use `len() != 0` with byte slices IMO, to avoid `nil` vs `empty slice` issue, especially if things come out of the DB. Same below with the asset ID bytes.